dao/src/main/java/org/thingsboard/server/dao/JpaDaoConfig.java

```diff
@@ -15,22 +15,70 @@
  */
 package org.thingsboard.server.dao;
 
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceProperties;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.orm.jpa.EntityManagerFactoryBuilder;
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.ComponentScan;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
 import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+import org.springframework.orm.jpa.JpaTransactionManager;
+import org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean;
+import org.springframework.transaction.PlatformTransactionManager;
 import org.springframework.transaction.annotation.EnableTransactionManagement;
+import org.thingsboard.server.dao.sqlts.sql.JpaSqlTimeseriesDao;
 import org.thingsboard.server.dao.util.TbAutoConfiguration;
 
+import jakarta.persistence.EntityManagerFactory;
+
+import javax.sql.DataSource;
+
 /**
  * @author Valerii Sosliuk
  */
 @Configuration
 @TbAutoConfiguration
-@ComponentScan({"org.thingsboard.server.dao.sql", "org.thingsboard.server.dao.attributes", "org.thingsboard.server.dao.cache", "org.thingsboard.server.cache"})
-@EnableJpaRepositories("org.thingsboard.server.dao.sql")
-@EntityScan("org.thingsboard.server.dao.model.sql")
+@ComponentScan({ "org.thingsboard.server.dao.sql", "org.thingsboard.server.dao.attributes",
+        "org.thingsboard.server.dao.cache", "org.thingsboard.server.cache" })
 @EnableTransactionManagement
+@EnableJpaRepositories(basePackages = {
+        "org.thingsboard.server.dao.sql" }, entityManagerFactoryRef = "entityManagerFactory", transactionManagerRef = "jpaTransactionManager")
+@EntityScan(basePackages = { "org.thingsboard.server.dao.model.sql", "org.thingsboard.server.dao.model.sqlts" })
 public class JpaDaoConfig {
```

Comment:
Buradaki değişikliklerin amacı neydi?

Reply:
Spring'de otomatik konfigurasyon var, sadece paket adını vererek otomatik konfigüre edebiliyoruz. 2 adet JPA kullanan bağlantımız (thingsboard ve readings) olduğu için hangi konfigurasyonun hangi repoları alacağını vermemiz lazım. [Pratik örnek](https://www.baeldung.com/spring-boot-configure-multiple-datasources), [Java spring'in veri tabanlarına bağlanması hakkında](https://youtu.be/mIFIb_JE47U?si=unNoyIOYeAZAaaa5&t=433)